unit/java_bytecode/java_bytecode_convert_method/convert_invoke_dynamic.cpp

```diff
@@ -0,0 +1,674 @@
+  const symbol_tablet &symbol_table,
+  const std::vector<codet> &instructions,
+  const lambda_assignment_test_datat &test_data,
+  const require_goto_statements::pointer_assignment_locationt
+  &lambda_assignment)
+{
+  const typecast_exprt &rhs_value = require_expr::require_typecast(
+    lambda_assignment.non_null_assignments[0].rhs());
+
+  const symbol_exprt &rhs_symbol =
+    require_expr::require_symbol(rhs_value.op0());
+
+  const irep_idt &tmp_object_symbol = rhs_symbol.get_identifier();
+
+  const auto tmp_object_assignments =
+    require_goto_statements::find_pointer_assignments(
+      tmp_object_symbol, instructions);
+
+  REQUIRE(tmp_object_assignments.non_null_assignments.size() == 1);
+  REQUIRE_FALSE(tmp_object_assignments.null_assignment.has_value());
+
+  const side_effect_exprt &side_effect_expr =
+    require_expr::require_side_effect_expr(
+      tmp_object_assignments.non_null_assignments[0].rhs(), ID_java_new);
+
+  const pointer_typet &lambda_temp_type =
+    require_type::require_pointer(side_effect_expr.type(), {});
+
+  const symbol_typet &lambda_implementor_type =
+    require_type::require_symbol(lambda_temp_type.subtype());
+
+  const irep_idt &tmp_class_identifier =
+    lambda_implementor_type.get_identifier();
+
+  const symbolt &lambda_implementor_type_symbol =
+    require_symbol::require_symbol_exists(symbol_table, tmp_class_identifier);
+
+  REQUIRE(lambda_implementor_type_symbol.is_type);
+  const class_typet &tmp_lambda_class_type =
+    require_type::require_complete_class(lambda_implementor_type_symbol.type);
+
+  REQUIRE(tmp_lambda_class_type.has_base(test_data.lambda_interface));
+  REQUIRE(tmp_lambda_class_type.has_base("java::java.lang.Object"));
+
+  class_hierarchyt class_hierarchy;
+  class_hierarchy(symbol_table);
+
+  const auto &parents = class_hierarchy.get_parents_trans(tmp_class_identifier);
+  REQUIRE_THAT(
+    parents,
+    Catch::Matchers::Vector::ContainsElementMatcher<irep_idt>{
+      test_data.lambda_interface});
+
+  const auto &interface_children =
+    class_hierarchy.get_children_trans(test_data.lambda_interface);
+
+  REQUIRE_THAT(
+    interface_children,
+    Catch::Matchers::Vector::ContainsElementMatcher<irep_idt>{
+      tmp_class_identifier});
+
+  const java_class_typet::componentt super_class_component =
+    require_type::require_component(tmp_lambda_class_type, "@java.lang.Object");
+
+  const symbol_typet &super_class_type = require_type::require_symbol(
+    super_class_component.type(), "java::java.lang.Object");
+
+  const symbolt &base_class_symbol = require_symbol::require_symbol_exists(
+    symbol_table, super_class_type.get_identifier());
+
+  REQUIRE(base_class_symbol.is_type);
+  const class_typet &super_class_type_struct =
+    require_type::require_incomplete_class(base_class_symbol.type);
+
+  require_type::require_component(super_class_type_struct, "@class_identifier");
+  // TODO verify the components of the class have been set correctly
```

Comment:
This todo will be aprt of the next tickets for lambdas? Once we know how exactly the code looks right, correct?

Reply:
Nope TODO no longer relevant as this is handled in a later transformation so not something we can test here. Removed the TODO